src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -654,4 +654,13 @@ public function getInvalidBinaryData()
             'misplaced equals character' => array('!!binary "SGVsbG8gd29ybG=Q"', '/The base64 encoded data \(.*\) contains invalid characters/'),
         );
     }
+
+    /**
+     * @expectedException \Symfony\Component\Yaml\Exception\ParseException
+     * @expectedExceptionMessage Malformed inline YAML string {this, is not, yaml}
+     */
+    public function testStringOffsetCastError()
+    {
+        Inline::parse('{this, is not, yaml}');
```

Comment:
Actually this is valid and should be equivalent to `{ this: ~, is not: ~, yaml: ~ }`.